### PR TITLE
Fix false positives for duplicate filters error when saving a new filter

### LINKED ISSUE #2937: [Bug]: Duplicate filter warning when creating filters that differ by amount inflow and amount outflow
### Verified issue does not already exist?

- [X] I have searched and found no existing issue
- [X] I will be providing steps how to reproduce the bug (in most cases this will also mean uploading a demo budget file)

### What happened?

When trying to create two filters with the following conditions, a duplicate filter warning is thrown and it cannot be saved:

Filter 1: Notes containing string, Amount **(inflow)** is greater than 0 
Filter 2: Notes containing string, Amount **(outflow)** is greater than 0

image
image

The filters are different, returns different results and should be allowed. Also if it is just a "warning" it should still allow the action, it is not an "error" condition.

### Where are you hosting Actual?

Docker

### What browsers are you seeing the problem on?

Microsoft Edge, Desktop App (Electron)

### Operating System

Windows 10

### PR DESCRIPTION
**Pull request by scivarolo**
Fixes #2937 

Adds additional logic to the `conditionExists` function to also compare the`options` object. Fields like Amount, Amount (inflow), Amount (outflow) were matching as the same condition because their only difference lies within the options object:

image

I also refactored the logic to remove Array.map() in favor of Array.some() so that the loops can short circuit as soon as the condition is met.

### CODE REVIEW

File: packages/loot-core/src/server/filters/app.ts [hunk 1]
```diff
@@ -56,50 +56,64 @@ async function filterNameExists(name, filterId, newItem) {
...
-              cond.value === fcond.value &&
-              cond.op === fcond.op &&
-              cond.field === fcond.field
-            );
-          }).length > 0;
-        fCondCheck = (i === 0 ? true : fCondCheck) && condCheck[i];
-        return true;
+      let allConditionsMatch = true;
+
+      conditions.some(cond => {
```
Comment (twk3) [inline packages/loot-core/src/server/filters/app.ts:71]:
```suggestion
      let allConditionsMatch = !conditions.some(cond => {
```

suggestion: eliminate the need to set the allConditionsMatch within this some function by just grabbing the result of `some`

**Review by twk3 — CHANGES_REQUESTED** [review]
@scivarolo  thanks for the fix! 

I've left one suggestion for your feedback

File: packages/loot-core/src/server/filters/app.ts (hunk 1 above)
Comment (scivarolo) [reply packages/loot-core/src/server/filters/app.ts]: Thanks for the suggestion! Took it a step further and simplified that chunk of logic even more.

**Review by MikesGlitch — APPROVED** [review]
lgtm! The file is missing some types but this fixes the bug. :shipit:

**Review by twk3 — APPROVED** [review]
Thanks @scivarolo !

**Pull request merged**